### PR TITLE
[Error] Add TaichiTypeError

### PR DESCRIPTION
**Pull request by lin-hitonami**
Related issue = #3938

### CODE REVIEW

File: python/taichi/lang/exception.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+from taichi._lib import core
```
Comment (ailzhang) [inline python/taichi/lang/exception.py:1]: How about `from taichi._lib.core import TaichiTypeError`?

**Review by ailzhang — APPROVED** [review]
LGTM!

File: python/taichi/lang/exception.py (hunk 1 above)
Comment (lin-hitonami) [reply python/taichi/lang/exception.py:1]: It seems that I can't do that... maybe because `taichi._lib.core` is a dynamic lib?

**Pull request merged**